jujugui/static/gui/src/app/components/entity-details/entity-details.js

```diff
@@ -216,13 +217,14 @@ YUI.add('entity-details', function() {
       this.refs.content.focus();
       // Be sure to convert the id to the legacy id as the URL will be in the
       // new id format.
-      this.detailsXhr = this.props.getEntity(this.props.id, this.fetchCallback);
+      const xhr = this.props.getEntity(this.props.id, this.fetchCallback);
+      this.xhrs.push(xhr);
     },
 
     componentWillUnmount: function() {
-      if (this.detailsXhr) {
-        this.detailsXhr.abort();
-      }
+      this.xhrs.forEach(xhr => {
+        xhr && xhr.abort && xhr.abort();
```

Comment:
Aren't we being too much defensive here? In what cases xhrs include an invalid xhr? I guess the answer is in the cases when the provided methods return undefined instead of what they are supposed to return, and I'd rather see these bugs exposed maybe?